spec/PeriodicBackgroundSync-index.bs

```diff
@@ -92,3 +94,95 @@ Fetch requests within the <code>periodicsync</code> event while [=in the backgro
+    1. If |timeTillScheduledEventForOrigin| is not null:
+        1. If |timeTillScheduledEventForOrigin| - |delayForOrigin| is greater than or equal to |minIntervalForOrigin|, abort these substeps.
+        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| and abort these substeps.
+        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|.
+    1. Let <var>minIntervalAcrossOrigins</var> be the minimum interval allowed between periodicSync events across origins. Let <var>timeSinceLastPeriodicSync</var> be the time since a periodicSync event was fired for any origin, if any, null otherwise.
+    1. If |timeSinceLastPeriodicSync| is null, set |timeToFire| to |delayForOrigin| + |now|.
+    1. Else:
+        1. If |timeSinceLastPeriodicSync| is greater than equal to |minIntervalAcrossOrigins|, set |timeToFire| to |delayForOrigin| + |now|.
+        1. Else, set <var>timeTillNextAllowedPeriodicSync</var> to |minIntervalAcrossOrigins| - |timeSinceLastPeriodicSync|. Set |timeToFire| to the maximum of |delayForOrigin| + |now|, and |timeTillNextAllowedPeriodicSync| + |now|.
+    1. Set the [=time to fire=] of |registration| to |timeToFire|.
+1. Else:
+    1. Let <var>countRetries</var> be the count of retries that have been made for this event.
+    1. Increment |countRetries|.
+    1. If |countRetries| is equal to the [=maximum number of retries=] allowed, set |countRetries| to 0 and follow the steps to [=first attempt of firing|fire the first attempt=].
+    1. Else, Set the [=time to fire=] of |registration| to |now| + a small back-off that is exponentially proportional to |countRetries|.
+
+API Description {#api-description}
+===================================
+## Extensions to the {{ServiceWorkerGlobalScope}} interface ## {#extensions-to-serviceworkerglobalscope}
+<script type="idl">
+partial interface ServiceWorkerGlobalScope {
+    attribute EventHandler onperiodicsync;
+};
+</script>
+
+## Extensions to the {{ServiceWorkerRegistration}} interface ## {#extensions-to-serviceworkerregistration}
+
+<script type="idl">
+[Exposed=(Window,Worker)]
+partial interface ServiceWorkerRegistration {
+  readonly attribute PeriodicSyncManager periodicSync;
+};
+</script>
+
+<div dfn-for="ServiceWorkerRegistration">
+A {{ServiceWorkerRegistration}} has a <dfn>periodic sync manager</dfn> (a {{PeriodicSyncManager}}), initially a new {{PeriodicSyncManager}} whose [=service worker registration=] is the [=context object=]'s [=service worker registration=].
+
+The <dfn attribute>periodicSync</dfn> attribute's getter must return the [=context object=]'s [=ServiceWorkerRegistration/periodic sync manager=].
```

Comment:
I's also update all other periodicSync references you had in the algorithm to point to this.

Reply:
They're different actually. The algorithm is referring to "periodicSync event". In this CL, when I introduce the periodicSync event, I have linked them:
https://github.com/mugdhalakhani/BackgroundSync/pull/7